### PR TITLE
fix for issue #60 (invalid bin_edges for SparselyBin)

### PR DESCRIPTION
**Pull request by twalen**

### CODE REVIEW

**Comment by sbrugman** [comment]
@twalen the repository now uses Github actions for running the tests. `testbasic.py` was renamed to `test_basic.py` so that `pytest` automatically picks it up. Could you please rebase?

**Comment by twalen** [comment]
@sbrugman is it ok now?

**Comment by sbrugman** [comment]
The git history now contains the recent commits to master. Would be better to have the single commit (easiest way is to use interactive rebase)

**Comment by twalen** [comment]
@sbrugman I've recreated the branch to have a single commit & push-forced.

**Comment by sbrugman** [comment]
Looks good now!

File: histogrammar/primitives/sparselybin.py [hunk 1]
```diff
@@ -672,41 +672,44 @@ def n_bins(self):
...
-            return 0
-        # trivial case
-        if low is None and high is None:
-            return (self.maxBin - self.minBin + 1)
-        # catch weird cases
-        elif low is not None and high is not None:
-            if low > high:
-                raise RuntimeError('low {low} greater than high {high}'.format(low=low, high=high))
-        # lowest edge
+            return [0, -1, 0, self.origin, self.origin+1]
```
Comment (sbrugman) [inline histogrammar/primitives/sparselybin.py:683]: Return type here is list, later it's a tuple (which is preferable for unpacking)

File: histogrammar/primitives/sparselybin.py [hunk 2]
```diff
@@ -672,41 +672,44 @@ def n_bins(self):
...
         else:
             maxBin = self.bin(high)
             if np.isclose(high, self.origin + self.bin_width() * maxBin):
                 maxBin -= 1
-        nbins = maxBin - minBin + 1
-        return nbins
+        numBins = maxBin + 1 - minBin
+        minBinLeftEdge = self.origin + self.bin_width() * minBin
+        maxBinRightEdge = self.origin + self.bin_width() * (maxBin + 1)
+        return (minBin, maxBin, numBins, minBinLeftEdge, maxBinRightEdge)
```
Comment (sbrugman) [inline histogrammar/primitives/sparselybin.py:698]: Parentheses not needed

**Pull request merged**